pkg/client/client.go

```diff
@@ -1873,6 +1893,26 @@ func (c *Client) RegisterConsolePlugin(ctx context.Context, name string) error {
 	return errors.Wrapf(err, "registering console-plugin %q with console %q failed", name, clusterConsole)
 }
 
+// ShouldEnableKSMCRSMetrics checks if kube-state-metrics' custom-resource-state-based metrics generation should be enabled.
+func (c *Client) ShouldEnableKSMCRSMetrics(ctx context.Context, lastKnownEnableKSMCRSMetricsState bool) bool {
```

Comment:
(nit) in general, the client package acts a wrapper to the Kube API and the operator package is in charge of interpreting the information, logging, ...
```suggestion
func (c *Client) VPACustomResourceDefinitionPresent(ctx context.Context) (bool, error) {
```